scripts/mod_loader/modapi/board.lua

```diff
@@ -1,96 +1,691 @@
+	elseif iTerrain == TERRAIN_RUBBLE then
+		
+		local rubbleState = CUtils.GetTileRubbleState(self, loc)
+		
+		if rubbleState == RUBBLE_BUILDING then
+			local hp_max = self:GetMaxHealth(loc)
+			self:SetBuilding(loc, hp_max, hp_max)
+			
+		else
+			self:SetTerrain(loc, TERRAIN_MOUNTAIN)
+		end
+	end
+end
+
+BoardClass.SetSnow = function(self, loc, snow)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "SetSnow",
+		params = { self, loc, snow },
+		{ "userdata|GameBoard&", "userdata|Point", "boolean|bool" }
+	}
+	
+	if snow == nil then
+		snow = true
+	end
+	
+	local custom_tile = self:GetCustomTile(loc)
+	
+	if snow then
+		if custom_tile == "" then
+			self:SetCustomTile(loc, "snow.png")
+		end
+	else
+		if custom_tile == "snow.png" then
+			self:SetCustomTile(loc, "")
+		end
+	end
+end
+
+BoardClass.SetUniqueBuilding = function(self, loc, buildingId)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "SetUniqueBuilding",
+		params = { self, loc, buildingId },
+		{ "userdata|GameBoard&", "userdata|Point", "string|string" }
+	}
+	
+	CUtils.TileSetUniqueBuilding(self, loc, buildingId)
+end
+
+BoardClass.GetUniqueBuilding = function(self, loc)
+	Tests.AssertSignature{
+		ret = "string",
+		func = "GetUniqueBuilding",
+		params = { self, loc },
+		{ "userdata|GameBoard&", "userdata|Point" }
+	}
+	
+	return CUtils.TileGetUniqueBuilding(self, loc)
+end
+
+BoardClass.RemoveUniqueBuilding = function(self, loc)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "RemoveUniqueBuilding",
+		params = { self, loc },
+		{ "userdata|GameBoard&", "userdata|Point" }
+	}
+	
+	CUtils.TileRemoveUniqueBuilding(self, loc)
+	self:SetBuilding(loc)
+end
+
+BoardClass.RemoveItem = function(self, loc)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "RemoveItem",
+		params = { self, loc },
+		{ "userdata|GameBoard&", "userdata|Point" }
+	}
+	
+	if not self:IsItem(loc) then
+		return
+	end
+	
+	CUtils.TileRemoveItem(self, loc)
+end
+
+BoardClass.GetItemName = function(self, loc)
+	Tests.AssertSignature{
+		ret = "string",
+		func = "GetItem",
+		params = { self, loc },
+		{ "userdata|GameBoard&", "userdata|Point" }
+	}
+	
+	if not self:IsItem(loc) then
+		return
+	end
+	
+	return CUtils.TileGetItemName(self, loc)
+end
+
+BoardClass.GetHighlighted = function(self)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "GetHighlighted",
+		params = { self },
+		{ "userdata|GameBoard&" }
+	}
+	
+	return CUtils.BoardGetHighlighted(self)
+end
+
+BoardClass.IsHighlighted = function(self, loc)
+	Tests.AssertSignature{
+		ret = "bool",
+		func = "IsHighlighted",
+		params = { self, loc },
+		{ "userdata|GameBoard&", "userdata|Point" }
+	}
+	
+	return CUtils.IsTileHighlighted(self, loc)
+end
+
+BoardClass.MarkGridLoss = function(self, loc, grid_loss)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "MarkGridLoss",
+		params = { self, loc, grid_loss },
+		{ "userdata|GameBoard&", "userdata|Point", "number|int" }
+	}
+	
+	CUtils.TileMarkGridLoss(self, loc, grid_loss)
 end
```

Comment:
❗ Calling this causes an error: `bad argument #4 to 'TileMarkGridLoss' (number expected, got no value)`  
I called: `Board:MarkGridLoss(Board:GetHighlighted(), 2)`, happens both over regular tiles as well as buildings, in test mech scenario

Reply:
Fixed in `06dd4ac`
Note that this is another function that must to be called each frame grid loss should be marked on a building.